src/Handlers/GeneralUiStateHandler.php

```diff
@@ -326,36 +330,18 @@ protected function _prepareLinksConfig($linksConfig)
      *
      * @since [*next-version*]
      *
-     * @param array|stdClass|Traversable $iterator The array or Traversable object to convert
+     * @param array|stdClass|Traversable $iterator The object to convert.
      *
-     * @throws InvalidArgumentException if $iterator is not an array or a Traversable object
+     * @throws InvalidArgumentException If $iterator is not iterable.
      *
      * @return array Result of iterator to array transform.
      */
     protected function _iteratorToArrayRecursive($iterator)
     {
-        if (!($iterator instanceof stdClass) && !is_array($iterator) && !$iterator instanceof Traversable) {
-            throw $this->_createInvalidArgumentException(
-                $this->__('Argument is not an array or stdClass or Traversable object'), null, null, $iterator
-            );
-        }
-        if ($iterator instanceof stdClass) {
-            return (array) $iterator;
-        }
-        if (method_exists($iterator, 'toArray')) {
-            return $iterator->toArray();
-        }
-        $array = [];
+        $iterator = $this->_normalizeIterable($iterator);
+        $array    = [];
         foreach ($iterator as $key => $value) {
-            if (is_scalar($value)) {
-                $array[$key] = $value;
-                continue;
-            }
-            if ($value instanceof Traversable) {
-                $array[$key] = $this->_iteratorToArrayRecursive($value);
-                continue;
-            }
-            if (is_array($value)) {
+            if ($value instanceof Traversable || is_array($value) || $value instanceof stdClass) {
```

Comment:
Redundant check. Cannot be anything else, since the value was normalized to iterable.

Reply:
No, at this stage value wasn't normalized to iterable. It will be normalized once it will be passed to the function.